### ISSUE TITLE
Attempts to read rdf file using wrong reader

### ISSUE THREAD

**Issue by elrayle** [ruby-rdf/rdf#353]
I am trying to read a remote file from worldcat.  This code worked with 2.2.3, but does not with 2.2.5.  Best I can trace, it is now trying to use the wrong reader for the filetype.  The content from worldcat is rdf+xml, but it is trying to use a TriX reader. 

The code I am using to read the graph is...
```
    graph = RDF::Graph.load('http://id.worldcat.org/fast/530369/rdf.xml')
    graph.size  # expect > 0
```

The major places I checked for problems are...
* lib/rdf/reader.rb:189 - self is RDF::TriX::Reader
* lib/rdf/util/file.rb:330 - remote_document reads in correctly and remote_document.string has the correct content from worldcat
* lib/rdf/reader.rb:195
```
options # {:base_uri=>"http://id.worldcat.org/fast/530369/rdf.xml", :headers=>{"Accept"=>"application/trix, */*;q=0.1"}}
## Accept is set to trix because reader is TriX
file.content_type # "application/rdf+xml"  ## which is the expected content type
```
* lib/rdf/reader.rb:215 - attempts to read rdf+xml with TriX reader returns 0 statements in the graph

**Comment by elrayle** [comment]
This cannot be tested through rspec since it denies attempts to make a real http call.  I put together a quick rails app to test this.

https://github.com/elrayle/rdf_issue_testing

**Comment by elrayle** [comment]
I noticed that two other URIs that I access do not have problems.

```
    graph = RDF::Graph.load('http://id.loc.gov/authorities/names/n79065220')
    graph.size  # PASS -- size > 0

    graph = RDF::Graph.load('http://aims.fao.org/aos/agrovoc/c_9513')
    graph.size  # PASS -- size > 0
```

So, I tried dropping off the `/rdf.xml` from the URI and that seems to work.  

```
    # graph = RDF::Graph.load('http://id.worldcat.org/fast/530369/rdf.xml')
    graph = RDF::Graph.load('http://id.worldcat.org/fast/530369')
    graph.size  # PASS -- size > 0 with rdf.xml removed
```

This allows my work to continue.  Still seems like a bug.

**Comment by gkellogg** [comment]
I'll need to see where it changed, but the code is settling on the TriX reader, because it accepts a file extension of ".xml", and RDF/XML does not. There is no other extension associated with TriX, which is the source of the problem. Previously, it must have deferred associating a reader until after downloading the file, and getting a Content-Type. This is likely due to an "Optimization" in RDF::Reader.for.

**Comment by gkellogg** [comment]
Released in 2.2.6.
